src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -702,6 +704,82 @@ private static FlowAnalysisAnnotations DecodeFlowAnalysisAttributes(PEModule mod
             return annotations;
         }
 
+#nullable enable
+        internal override ImmutableArray<int> InterpolatedStringHandlerArgumentIndexes
+        {
+            get
+            {
+                ImmutableArray<int> indexes = _lazyInterpolatedStringHandlerAttributeIndexes;
+                if (indexes.IsDefault)
+                {
+                    indexes = DecodeInterpolatedStringHandlerArgumentAttribute();
+                    Debug.Assert(!indexes.IsDefault);
+                    var initialized = ImmutableInterlocked.InterlockedInitialize(ref _lazyInterpolatedStringHandlerAttributeIndexes, indexes);
+                    Debug.Assert(initialized || indexes.SequenceEqual(_lazyInterpolatedStringHandlerAttributeIndexes));
+                }
+
+                return indexes;
+            }
+        }
+
+        private ImmutableArray<int> DecodeInterpolatedStringHandlerArgumentAttribute()
+        {
+            // If we encounter an invalid attribute, we ignore the contents and just use the empty list.
+
+            if (Type is not NamedTypeSymbol { IsInterpolatedStringHandlerType: true })
+            {
+                return ImmutableArray<int>.Empty;
+            }
+
+            var paramNames = _moduleSymbol.Module.GetInterpolatedStringHandlerArgumentAttributeValues(_handle);
+            Debug.Assert(!paramNames.IsDefault);
+
+            if (paramNames.IsEmpty)
+            {
+                return ImmutableArray<int>.Empty;
+            }
+
+            var builder = ArrayBuilder<int>.GetInstance(paramNames.Length);
+            var parameters = ContainingSymbol switch
+            {
+                MethodSymbol { Parameters: var p } => p,
+                PropertySymbol { Parameters: var p } => p,
+                _ => throw ExceptionUtilities.UnexpectedValue(ContainingSymbol)
+            };
+
+            foreach (var name in paramNames)
+            {
+                switch (name)
+                {
+                    case null:
+                    case { Length: 0 } when ContainingSymbol.IsStatic:
```

Comment:
>when ContainingSymbol.IsStatic: [](http://example.com/codeflow?start=39&length=31)

I guess I am missing what static-ness of a container has to do with validity of the data? Is there a relevant spec?

Reply:
Perhaps the Length check confused me as I thought we are dealing with an array. Consider comparing with an epty string instead, for clarity. Below as well.